tests/framework/components/test_parser.py

```diff
@@ -126,7 +126,7 @@ def test_parse_and_prep_components(parser):
     prepped_components = [
         parser.prep_component(component) for component in TEST_COMPONENTS_PARSED
     ]
-
+    prepped_components = [(x, tuple(y)) for (x, y) in prepped_components]
```

Comment:
why not just change TEST_COMPONENTS_PREPPED?

Reply:
We can't convert it into a set if we use lists instead of tuples